# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: mobile/android/geckoview/src/androidTest/java/org/mozilla/geckoview/test/AccessibilityTest.kt
# Commit: 57434a88c266
# Full Hash: 57434a88c266e90a7f4448f87c3c8ba882e383d1
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-11-29 21:47:49
# Description:
#   Bug 1801879: Don't cross document boundaries in nsAccUtils::GetSelectableContainer. r=eeejay,geckoview-reviewers,owlish
#   
#   Trying to access a local OuterDocAccessible from the Android UI thread was causing a crash.
#   We shouldn't be crossing document boundaries anyway.
#   
# ==============================================================================

diff -r 58bf4c34d5c2 -r 57434a88c266 mobile/android/geckoview/src/androidTest/java/org/mozilla/geckoview/test/AccessibilityTest.kt
--- a/mobile/android/geckoview/src/androidTest/java/org/mozilla/geckoview/test/AccessibilityTest.kt	Tue Nov 29 05:15:07 2022 +0000
+++ b/mobile/android/geckoview/src/androidTest/java/org/mozilla/geckoview/test/AccessibilityTest.kt	Tue Nov 29 05:18:23 2022 +0000
@@ -1056,6 +1056,18 @@
 
         provider.performAction(nodeId, AccessibilityNodeInfo.ACTION_SELECT, null)
         waitUntilSelect(false)
+
+        // Ensure that querying an option outside of a selectable container
+        // doesn't crash (bug 1801879).
+        mainSession.evaluateJS("document.getElementById('outsideSelectable').focus()")
+        sessionRule.waitUntilCalled(object : EventDelegate {
+            @AssertCalled(count = 1)
+            override fun onFocused(event: AccessibilityEvent) {
+                nodeId = getSourceId(event)
+                val node = createNodeInfo(nodeId)
+                assertThat("Focused outsideSelectable", node.text.toString(), equalTo("outside selectable"))
+            }
+        })
     }
 
     @Test fun testMutation() {
